### PR TITLE
Wiremock

### PR DESCRIPTION
**Pull request by MrCustomizer**
I've added test cases, which cover most of the http calls of the classes, which extend from AMatrixHttpClient.

Could you please have a look at it and tell me, if this can be merged or if I should make some adjustments, first?

### CODE REVIEW

**Comment by MrCustomizer** [comment]
The regex fix in the class MatrixHttpContent, we've talked about yesterday, is also included.

File: src/test/java/io/kamax/matrix/client/MatrixHttpRoomTest.java [hunk 1]
```diff
@@ -0,0 +1,297 @@
...
+        stubFor(get(urlEqualTo(nameUrl)).willReturn(aResponse().withStatus(403).withBody(error403Response)));
+
+        try {
+            createRoomObject().getName();
+        } catch (MatrixClientRequestException e) {
+            // TODO getName does not throw Exceptions with error infos, yet
+            // checkErrorInfo403(e);
+            return;
+        }
+        fail("In this case, an exception has to be thrown.");
```
Comment (maxidor) [inline src/test/java/io/kamax/matrix/client/MatrixHttpRoomTest.java]: Use `@Test(expected = ....)`

File: src/test/java/io/kamax/matrix/client/MatrixHttpRoomTest.java [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+        stubFor(get(urlEqualTo(nameUrl)).willReturn(aResponse().withStatus(429).withBody(error403Response)));
+
+        try {
+            createRoomObject().getName();
+        } catch (MatrixClientRequestException e) {
+            // TODO getName does not throw Exceptions with error infos, yet
+            // checkErrorInfo429(e);
+            return;
+        }
+        fail("In this case, an exception has to be thrown.");
```
Comment (maxidor) [inline src/test/java/io/kamax/matrix/client/MatrixHttpRoomTest.java]: Use @Test(expected = ....)

File: src/test/java/io/kamax/matrix/client/MatrixHttpRoomTest.java [hunk 3]
```diff
@@ -0,0 +1,297 @@
...
+    public void getTopicError403() throws URISyntaxException {
+        stubFor(get(urlEqualTo(topicUrl)).willReturn(aResponse().withStatus(403).withBody(error403Response)));
+
+        try {
+            createRoomObject().getTopic();
+        } catch (MatrixClientRequestException e) {
+            checkErrorInfo403(e);
+            return;
+        }
+        fail("In this case, an exception has to be thrown.");
```
Comment (maxidor) [inline src/test/java/io/kamax/matrix/client/MatrixHttpRoomTest.java]: This should go after `createRoomObject()...` in the `try {}`.

File: src/test/java/io/kamax/matrix/client/MatrixHttpRoomTest.java [hunk 4]
```diff
@@ -0,0 +1,297 @@
...
+
+    @Test
+    public void getTopicError403() throws URISyntaxException {
+        stubFor(get(urlEqualTo(topicUrl)).willReturn(aResponse().withStatus(403).withBody(error403Response)));
+
+        try {
+            createRoomObject().getTopic();
+        } catch (MatrixClientRequestException e) {
+            checkErrorInfo403(e);
+            return;
```
Comment (maxidor) [inline src/test/java/io/kamax/matrix/client/MatrixHttpRoomTest.java]: Not needed if we refactor the `fail()` (see next comment)

**Pull request merged**